### ISSUE TITLE
Allow certain response headers coming from AzurePipelinesCredential to be logged

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-for-rust#1823]
These two response headers are useful for the service team to debug and diagnose issues, and hence it is requested that we do log them.
- x-vss-e2eid
- x-msedge-ref

We decided, other than logging them, to also add these header values to the exception message that is thrown, so they are always visible, even with logging disabled.

Given our logged headers are based on an allow list, currently these two headers are REDACTED.

https://github.com/Azure/azure-sdk-for-net/blob/a70122eed52b0980afc10db30c76cb56b7bb76db/sdk/core/Azure.Core/src/DiagnosticsOptions.cs#L44-L71

**Comment by ahsonkhan** [comment]
See https://github.com/Azure/azure-sdk-for-cpp/pull/6001 for an example change from C++.

**Comment by chlowell** [comment]
Moving this to the backlog because it's blocked on an allow-list API for header logging (#682) and is a nice-to-have but not required feature. (We include the headers in question in error messages; logging them would make them more visible, however they aren't invisible otherwise.)
